packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -361,62 +361,102 @@ it('should handle checkbox clicks', async () => {
   expect(onRowClick).toHaveBeenCalledTimes(2);
 
-  await userEvent.click(getByText(mockedData()[1].name)); // Reselect; lastSelectedRowId = undefined -> 1
-  expect(rows[0].classList).not.toContain('iui-selected');
+  // Shift click special case test #2
+  // When a row is clicked before shift click (lastSelectedRowId), selection starts from that row and ends at the currently clicked row
+  // But if the startIndex > endIndex, then startIndex and endIndex are swapped
+  // (Here startIndex = 1, endIndex = 0)
+  await user.keyboard('{Shift>}'); // Hold Shift
+  await user.click(getByText(data[0].name));
+
+  expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).toContain('iui-selected');
   expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[3].classList).not.toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(3);
   expect(onRowClick).toHaveBeenCalledTimes(3);
 
-  await user.keyboard('[ControlLeft>]'); // Press Control (without releasing it)
-  await user.click(getByText(mockedData()[2].name)); // Perform a click with `ctrlKey: true`
-  expect(rows[0].classList).not.toContain('iui-selected');
+  await user.keyboard('{/Shift}{Control>}'); // Release Shift & Hold Control
+  await user.click(getByText(data[3].name)); // lastSelectedRowId = 1 -> 3 (Ctrl click updates lastSelectedRowId)
+  expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).toContain('iui-selected');
-  expect(rows[2].classList).toContain('iui-selected');
+  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[3].classList).toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(4);
   expect(onRowClick).toHaveBeenCalledTimes(4);
 
-  // Shift click test #2
-  // When a row is clicked before shift click (lastSelectedRowId), selection starts from that row and ends at the currently clicked row
-  // But if the startIndex > endIndex, then startIndex and endIndex are swapped
-  await user.keyboard('[/ControlLeft][ShiftLeft>]'); // Release Ctrl and Press Shift (without releasing it)
-  await user.click(getByText(mockedData()[0].name)); // Perform a click with `shiftKey: true`
+  await user.keyboard('{/Control}{Shift>}'); // Release Control & Hold Shift
+  await user.click(getByText(data[4].name));
 
-  expect(rows[0].classList).toContain('iui-selected');
-  expect(rows[1].classList).toContain('iui-selected');
+  expect(rows[0].classList).not.toContain('iui-selected');
+  expect(rows[1].classList).not.toContain('iui-selected');
   expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[3].classList).toContain('iui-selected');
+  expect(rows[4].classList).toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(5);
   expect(onRowClick).toHaveBeenCalledTimes(5);
+
+  const checkboxes = container.querySelectorAll<HTMLInputElement>(
+    '.iui-table-body .iui-checkbox',
+  );
+  expect(checkboxes.length).toBe(5);
```

Comment:
Maybe move this checkbox check to be before or after the series of shift/control/regular clicking? It seems kinda random to be here to me

Reply:
I had kept it here just to confirm that there are indeed five checkboxes since there are five rows.

But the number of checkboxes would not change after shift/control/regular clicks. So, I think doing this check just once in the current line location is good.

Reply:
Ah, I don't know that I noticed that 'checkboxes' was used down below \/. This is fine